seed/ts-sdk/basic-auth-environment-variables/src/auth/BasicAuthProvider.ts

```
@@ -0,0 +1,38 @@
+// This file was auto-generated by Fern from our API Definition.
+
+import * as core from "../core/index.js";
+
+export namespace BasicAuthProvider {
+    export interface Options {
+        username?: core.Supplier<string | undefined>;
+        accessToken?: core.Supplier<string | undefined>;
+    }
+}
+
+export class BasicAuthProvider implements core.AuthProvider {
+    private readonly username: core.Supplier<string | undefined> | undefined;
+    private readonly password: core.Supplier<string | undefined> | undefined;
+
+    constructor(options: BasicAuthProvider.Options) {
+        this.username = options.username;
+        this.password = options.accessToken;
+    }
+
+    public async getAuthRequest(): Promise<core.AuthRequest> {
```

We should throw the same errors as before:
```ts
        const username = (await core.Supplier.get(this._options.username)) ?? process?.env.USERNAME;
        if (username == null) {
            throw new errors.SeedBasicAuthEnvironmentVariablesError({
                message:
                    "Please specify a username by either passing it in to the constructor or initializing a USERNAME environment variable",
            });
        }

        const accessToken = (await core.Supplier.get(this._options.accessToken)) ?? process?.env.PASSWORD;
        if (accessToken == null) {
            throw new errors.SeedBasicAuthEnvironmentVariablesError({
                message:
                    "Please specify a accessToken by either passing it in to the constructor or initializing a PASSWORD environment variable",
            });
        }
```